src/rhsm/utils.py

```diff
@@ -23,7 +23,25 @@
 
 import rhsm.config
 from rhsm.config import DEFAULT_PROXY_PORT
-import subscription_manager.injection as inj
+
+try:
+    import subscription_manager.injection as inj
+except ImportError:
+
+    class inj:
+        """Fake injection used for progress messages.
+
+        When subscription-manager is not installed, the import will fail.
+        This fallback solution disables the progress messages completely;
+        that feature is primarily targeted at subscription-manager, but has been
+        placed into rhsm package for technical reasons.
+        """
+
+        PROGRESS_MESSAGES: str = "PROGRESS_MESSAGES"
+
+        @classmethod
+        def require(cls, *args, **kwargs) -> bool:
+            return False
```

Comment:
And returning `cls.enabled` here?

In that case it could be possible to enable progress messages without subscription-manager module in other applications.